src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/HttpHandlerDiagnosticListener.cs

```diff
@@ -557,33 +558,69 @@ private HttpHandlerDiagnosticListener() : base(DiagnosticListenerName)
 
         private void RaiseRequestEvent(HttpWebRequest request)
         {
-            // If System.Net.Http.Request is on, raise the event
-            if (this.IsEnabled(RequestWriteName))
+            if (request.Headers.Get(RequestIdHeaderName) != null)
             {
-                long timestamp = Stopwatch.GetTimestamp();
-                this.Write(RequestWriteName,
-                    new
+                // this request was instrumented by previous RaiseRequestEvent
+                return;
+            }
+
+            if (this.IsEnabled(ActivityName, request))
+            {
+                var activity = new Activity(ActivityName);
+
+                // Only send start event to users who subscribed for it, but start activity anyway
+                if (this.IsEnabled(RequestStartName))
+                {
+                    long timestamp = Stopwatch.GetTimestamp();
```

Comment:
I'd be happy to do this change and use GetUtcNow in Activity.Start() method is StartTime is not provided.

However in this particular case it does not seem useful:
Activity.Current does **not** flow to the `RaiseResponseEvent` (to be precise, sometimes it flows, sometimes does not). I've spent some time analyzing HttpWebRequest code, but do not have good answer why it happens.
So Activity is stopped right away after headers are injected.

In the `RaiseResponseEvent`, since there is no Activity.Current, we have to send timestamp anyway and it makes sense to to the same in `RaiseRequestEvent`.